lib/src/QGVWidgetScale.cpp

```diff
@@ -77,11 +80,49 @@ Qt::Orientation QGVWidgetScale::getOrientation() const
 
 QString QGVWidgetScale::getDistanceLabel(int meters, int accuracy) const
 {
-    if (meters > 1000) {
-        return tr("%1 km").arg(QString::number(static_cast<double>(meters) / 1000, 'f', accuracy));
-    } else {
-        return tr("%1 m").arg(QString::number(static_cast<double>(meters), 'f', accuracy));
+    QString result = "";
+    if ( mDistanceUnits == DistanceUnits::Kilometers) {
+        if (mUseMetersForSmallDistance) {
+            if (meters > 1000) {
+                result = tr("%1 km").arg(QString::number(static_cast<double>(meters) / 1000, 'f', accuracy));
+            } else {
+                result = tr("%1 m").arg(QString::number(static_cast<double>(meters), 'f', accuracy));
+            }
+        } else {
+            result = tr("%1 m").arg(QString::number(static_cast<double>(meters), 'f', accuracy));
+        }
+    } else if (mDistanceUnits == DistanceUnits::NauticalMiles) {
+        double nauticalMiles = meters / 1852.0;
```

Comment:
I know it was hard-coded before, but now I see benefit to have convert functions, like meterToKilometer, meterToMiles, etc in QGVGlobal.h
I think it could be useful for future usage